Return Turtle Content

MattTennison opened the pull request:
# What?

This PR will change the /content endpoint to produce turtle content.

# Why?

Related Issue with reasonings: https://github.com/Multi-User-Domain/mud-jena/issues/1

MattTennison committed:
Add IntelliJ folder to .gitignore

MattTennison committed:
Format ContentController

Using Prettier Java VSCode Extension

MattTennison committed:
Refactor ContentController

MattTennison committed:
Bump to Maven JDK to 1.8 to use Lambda expressions

MattTennison committed:
Add senses resource type and sight property

These are currently only available in my vocab fork, will become part of
main vocab repo once I join the Multi-User-Domain org

MattTennison committed:
Produce turtle output in content endpoint

calummackervoy commented on src/main/java/com/mackervoy/calum/mud/content/ContentController.java:
```diff
@@ -1,59 +1,69 @@
+
+		return getDescriber(safeUri, m)
+			.map(describer -> describer.describe(m, safeUri))
+      .map(description -> descriptionToSensesModel(description))
```
I think this stage of the pipeline should be managed by the `describer.describe` because  some describers might want to extend/replace the senses model, others not ?

MattTennison replied:
Ah good idea - so tweak the describer interace to return a sense model?

calummackervoy replied:
Yeah I think so?

MattTennison replied:
I've made that change here: https://github.com/Multi-User-Domain/mud-jena/pull/7/commits/25f9538e247a2c2252c3bf3951527680b3abdaf2 😄

calummackervoy replied:
Looks good!

calummackervoy commented on src/main/java/com/mackervoy/calum/mud/content/ContentController.java:
```diff
@@ -1,59 +1,69 @@
+			.orElse(null);
+  }
+	
+	private Optional<IContentDescriber> getDescriber(String uri, Model m) {
```
really nice!

calummackervoy commented on src/main/java/com/mackervoy/calum/mud/content/ContentController.java:
```diff
@@ -1,59 +1,69 @@
+    return m;
+  }
+
+  private String modelToTurtle(Model m) {
```
Good shout - we should move this to a universal utility function at some point or some sort of abstract class for Controllers to extend

calummackervoy reviewed (changes requested):
Thanks!

MattTennison committed:
Return senses model from describer instead of visual description string

As per https://github.com/Multi-User-Domain/mud-jena/pull/7/files/c84d5096436d13ad5861891e4633497153416c0d#r575858555

MattTennison committed:
Return Optional<Model> instead of Model from IContentDescriber

As some describers won't be able to describe certain content, and they were returning
null. This change helps make that behaviour more explicit in the type signature.

calummackervoy commented:
I guess there's a vocabulary PR to open and some React changes ... interesting question how to do the second part of that... I guess we can just fetch the content from the `mudcontent:sees` property for the MR ? Interesting because we'll need to start implementing a system responsible for reading returned content and deciding what to display from it, when to look for more content etc

calummackervoy committed:
Merge branch 'master' into turtle-content

MattTennison committed:
Rename senses to content

To match this change: https://github.com/MattTennison/vocab/commit/a0ebdad733be7b477accaa651b821d44a607fb88

MattTennison commented:
> I guess there's a vocabulary PR to open and some React changes ... interesting question how to do the second part of that... I guess we can just fetch the content from the `mudcontent:sees` property for the MR ? Interesting because we'll need to start implementing a system responsible for reading returned content and deciding what to display from it, when to look for more content etc

I've made changes to the vocab repo, but had difficulty pushing them to the MUD version (permissions issues, not sure why). I forked it and made a PR here: https://github.com/Multi-User-Domain/vocab/pull/2

I haven't made any changes to the React code, I'd be up for making those but it might be a job for next weekend. If that's okay, I'll raise an issue and assign it to myself, if not - happy to review a PR from you 😄

MattTennison committed:
Rename senses to content in code

To match RDF model

MattTennison committed:
Change the final senses references to content

Done a search through the codebase now, because I kept missing things!

calummackervoy commented:
OK nice one thanks! Of course no worries, thanks for doing this :) I'll take on the React code, I have a clear idea of what to do in my mind so piece of cake :)

MattTennison committed:
Extract content to seperate constants vocab file

To match change in: https://github.com/Multi-User-Domain/vocab/pull/2/commits/1a6f2861f31eea0fef6de765ad814c5708eec396

calummackervoy commented:
If https://github.com/Multi-User-Domain/mud-react/pull/15 is alright then we're ready to merge :)
